### DISCUSSION TITLE
Project/Dataset Autopopulation

### DISCUSSION THREAD

**Discussion by emilyernst** [CafIncubator/Midden#106] [Ideas]
Discussion from today's meeting brought up the possibility of autopopulating fields such as:

- contacts field on projects become drop down menu for a dataset
- projects dropdown menu for datasets

Are there any other fields that could become autopopulated?

**Comment by bryanrcarlson** [comment]
Having project information from the catalog.json file populate the input fields in the dataset editor has a lot of benefits. Providing these "benefits" also pushes the workflow towards best practices (define your project before you define datasets; hopefully meaning you create metadata for the project BEFORE you collect data). 

I can also see:
- Variables auto-populating from other datasets within the same project (assuming uniform variable names)
- Tags same as variables
- Spatial Extent auto-populating, although this already has a dropdown menu from the templates in app-config.json
- Temporal Extend auto-populating
- Parent Datasets dropdown from other datasets in the same Project (or all of Midden, actually)

Actually, the above list is mostly related to grouping datasets within project and not the project-metadata itself. One reasonable alternative to more project-specific metadata might be to mine the catalog.json file to auto-populate drop-down menus with the top values being those in the same project. So, for the contact field example, instead of defining contacts in the project editor (or manually in YAML front-matter), the user will need to define at least one dataset with the same project name then additional datasets will have those same contacts as values in a drop-down menu.

It makes more sense, and is potentially more user-friendly, to define the metadata with the project, as @emilyernst proposed. But the alternative outlined above allows the DESCRIPTION.md files to stay cleaner and keeps the schemas simpler.
